src/sparseml/keras/optim/modifier_pruning.py

```diff
@@ -222,10 +228,8 @@ def target_sparsity(self, step: int, tensor=None) -> float:
         if tensor is None:
             raise ValueError("Invalid empty tensor")
         if self._start_step <= step < self._end_step:
-            mask = tensorflow.cast(tensorflow.not_equal(tensor, 0.0), tensor.dtype)
-            sparsity = tensorflow.math.reduce_sum(1.0 - mask).numpy() / tensorflow.size(
-                tensor
-            )
+            mask = tf.cast(tf.not_equal(tensor, 0.0), tensor.dtype)
+            sparsity = tf.math.reduce_sum(1.0 - mask).numpy() / tf.size(tensor)
```

Comment:
this is going to return a numpy float and not a python float. Need to update the types or use .item.

Additionally, the sparsity is set to None and returned and the return type does not reflect that for this function